core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -180,10 +184,17 @@ public static boolean hasClass(MavenProject project, String ... classNames) {
     }
 
     /**
-     * Returns the spring boot `application.properties` or an empty properties object if not found
+     * Returns the spring boot configuration (supports `application.properties` and `application.yml`)
+     * or an empty properties object if not found
      */
     public static Properties getSpringBootApplicationProperties(MavenProject project) {
-        return getPropertiesFile(project, "application.properties");
+        URLClassLoader compileClassLoader = getCompileClassLoader(project);
```

Comment:
Yeah, I tried to stick with the existing code, but I'm ok to try a refactoring :)